spec/classes/foreman_proxy__plugin__dynflow_spec.rb

```diff
@@ -66,6 +66,28 @@
         end
       end
 
+      describe 'with open file limits defined' do
+        let :facts do
+          facts
+        end
+
+        let :pre_condition do
+           "include foreman_proxy"
+        end
+
+        let :params do {
+          :open_file_limit => 8000,
+        } end
+
+        if has_core
+          it { should contain_foreman_proxy__plugin('dynflow_core') }
+          it { should contain_systemd__service_limits('smart_proxy_dynflow_core.service').with_limits({'LimitNOFILE' => 8000}) }
+        else
+          it { should_not contain_foreman_proxy__plugin('dynflow_core') }
+          it { should_not contain_systemd__service_limits('smart_proxy_dynflow_core.service').with_limits({'LimitNOFILE' => 8000}) }
```

Comment:
It's better to remove the ```with_limits``` here. That way you test that the service limits are not present at all. Otherwise they could be there but with a different value.